packages/taquito/src/wallet/interface.ts

```diff
@@ -34,10 +45,22 @@ export interface WalletProvider {
   /**
    * @description Transform WalletIncreasePaidStorageParams into a format compliant with the underlying wallet
    */
-  mapIncreasePaidStorageWalletParams: (params: () => Promise<WalletIncreasePaidStorageParams>) => Promise<any>;
+  mapIncreasePaidStorageWalletParams: (
+    params: () => Promise<WalletIncreasePaidStorageParams>
+  ) => Promise<any>;
 
   /**
    * @description Request the wallet to send an operation batch
    */
   sendOperations: (params: any[]) => Promise<string>;
+
+  /**
+   * @description Request the wallet to sign a payload
+   */
+  sign(signingRequest: { payload: string; watermark?: Uint8Array }): Promise<string>;
+
+  /**
+   * @description Get the public key from the wallet
+   */
+  getPublicKey(): Promise<string | undefined>;
```

Comment:
might not have the context, will this ever get undefined cuz get pkh doesn't that made me wonder should we change the pkh too or keep pkh and make this return string for sure